jiant/scripts/postproc/xtreme/xtreme_submission.py

```diff
@@ -0,0 +1,413 @@
+        type=str,
+        default=None,
+        help="Path to val_metrics.json for bucc2018. Contains the optimal threshold,"
+        " to be used for generating test predictions.",
+    )
+
+    # === Model parameters === #
+    model_type = zconf.attr(type=str, required=True)
+    model_path = zconf.attr(type=str, required=True)
+    model_config_path = zconf.attr(default=None, type=str)
+    model_tokenizer_path = zconf.attr(default=None, type=str)
+    model_load_mode = zconf.attr(default="from_ptt", type=str)
+    model_save_mode = zconf.attr(default="all", type=str)
+
+    # === Training Learning Parameters === #
+    learning_rate = zconf.attr(default=1e-5, type=float)
+    adam_epsilon = zconf.attr(default=1e-8, type=float)
+    max_grad_norm = zconf.attr(default=1.0, type=float)
+    optimizer_type = zconf.attr(default="adam", type=str)
+
+    # Specialized config
+    no_cuda = zconf.attr(action="store_true")
+    fp16 = zconf.attr(action="store_true")
+    fp16_opt_level = zconf.attr(default="O1", type=str)
+    local_rank = zconf.attr(default=-1, type=int)
+    server_ip = zconf.attr(default="", type=str)
+    server_port = zconf.attr(default="", type=str)
+    force_overwrite = zconf.attr(action="store_true")
+    seed = zconf.attr(type=int, default=-1)
+
+
+def run_loop(args: RunConfiguration):
+    quick_init_out = initialization.quick_init(args=args, verbose=True)
+    with quick_init_out.log_writer.log_context():
+        if args.jiant_task_container_path:
+            jiant_task_container = container_setup.create_jiant_task_container(
+                **py_io.read_json(args.jiant_task_container_path)
+            )
+        else:
+            raise RuntimeError("Need `jiant_task_container_path` or individual config paths")
+        runner = setup_runner(
+            args=args,
+            jiant_task_container=jiant_task_container,
+            quick_init_out=quick_init_out,
+            verbose=True,
+        )
+    supertask, output_dir = args.supertask, args.output_dir
+    if supertask in ["xnli", "pawsx"]:
+        write_preds_for_classification(
+            runner=runner,
+            supertask=supertask,
+            output_dir=output_dir,
+            skip_if_done=args.skip_if_done,
+        )
+    elif supertask in ["udpos", "panx"]:
+        write_preds_for_tagging(
+            runner=runner,
+            supertask=supertask,
+            output_dir=output_dir,
+            skip_if_done=args.skip_if_done,
+        )
+    elif supertask in ["xquad", "mlqa"]:
+        write_preds_for_qa(
+            runner=runner,
+            supertask=supertask,
+            output_dir=output_dir,
+            phase="test",
+            skip_if_done=args.skip_if_done,
+        )
+    elif supertask == "tydiqa":
+        write_preds_for_qa(
+            runner=runner,
+            supertask="tydiqa",
+            output_dir=output_dir,
+            phase="val",
+            skip_if_done=args.skip_if_done,
+        )
```

Comment:
This way of generating predictions differs from the current way of generating formatted predictions (as part of the Task). Are there constraints preventing this from following the current standard?

Reply:
Let me look into moving this over.

Reply:
Unlike the GLUE/SuperGLUE tasks, in XTREME we're generating and writing the predictions in the script. I've renamed the functions to reflect this.